src/main/java/gregtech/api/items/toolitem/EnchantmentLevel.java

```diff
@@ -0,0 +1,15 @@
+package gregtech.api.items.toolitem;
+
+public class EnchantmentLevel {
+    private double level;
+    private double levelGrowth;
+
+    public EnchantmentLevel(double level, double levelGrowth) {
+        this.level = level;
+        this.levelGrowth = levelGrowth;
+    }
+
+    public int getLevel(int harvestTier) {
+        return (int) Math.min(Byte.MAX_VALUE, Math.floor(this.level + this.levelGrowth * harvestTier));
```

Comment:
We probably should not bound the enchantment level at one end by Byte.MAX_VALUE. What is the maximum level allowed by the vanilla system?

Reply:
The "bigger" hard limit for enchantments are Short.MAX which comes from data type used to encode enchant levels in NBT. The "smaller" hard limit enforced in `ItemStack#addEnchantment` is Byte.MAX.

Reply:
If we don't want to go to the point where enchantments start having strange effects due to having large values, Byte.MAX is best for this situation.